packages/react/src/Pagination/PaginationDesktop.client.tsx

```diff
@@ -0,0 +1,58 @@
+import "@axa-fr/design-system-css/dist/Pagination/PaginationDesktop.client.scss";
+import { FC } from "react";
+
+type Props = {
+  numberPage: number;
+  currentPage: number;
+  setCurrentPage: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const PaginationDesktop: FC<Props> = ({
+  numberPage,
+  currentPage,
+  setCurrentPage,
+}) => {
+  const sevenPage = 7;
+  const lastFourPage = 4;
+  const fivePage = 5;
+  const pages = Array.from({ length: numberPage }, (_, index) => index + 1);
+  let filteredNumbers = pages;
```

Comment:
Delete filteredNumbers and use useState for pages.
Add a useEffect to update his value when `numberPage` is updated.

Reply:
j'ai besoin de filteredNumbers  et de pages. Car pages me permet d'avoir toute les pages et filteredNumbers  me permet de filtrer seulement les pages que je souhaite afficher.

Reply:
un useState / useEffect changerait quelque chose ici ?
numberPage est passée en props si elle change le composant sera rerender et tout sera bien mis à jour

Reply:
> un useState / useEffect changerait quelque chose ici ? numberPage est passée en props si elle change le composant sera rerender et tout sera bien mis à jour

Oui mais à chaque fois que `currentPage` sera mis à jour il va tout recalculer sans que ce soit nécessaire.
Autant recalculer `pages` uniquement quand le nombre de pages change.

Reply:
ouais le currentPage qui change va tout faire recalculer, autant tout bien protéger si ça peut éviter des rerender dans tous les sens